research/conditional/utils/conditional_trainer.py

```diff
@@ -248,13 +251,24 @@ def _eval_step(self, step: int):
         if self.should_evaluate_dynamic_groupsize:
             batches = list(batches)
             contmoe_layers = [
-                l for _, l in self.layer_manager._layers if isinstance(l, ContinuousMoE)
+                l
+                for _, l in self.layer_manager._layers
+                if isinstance(l, (ContinuousMoE, ExpertChoiceFF))
```

Comment:
how can l be an instance of a tuple?

Reply:
https://stackoverflow.com/questions/33311258/python-check-if-variable-isinstance-of-any-type-in-list